pkg/cacheutil/memcached_client.go

```diff
@@ -88,6 +88,14 @@ type memcachedClientBackend interface {
 	Set(item *memcache.Item) error
 }
 
+// updatableServerSelector extends the interface used for picking a memcached server
+// for a key to allow servers to be updated at runtime.
+type updatableServerSelector interface {
+	memcache.ServerSelector
+
+	SetServers(servers ...string) error
```

Comment:
This is only for Cortex/Mimir, right? What's the use case behind this function?

Reply:
The use case was to allow me to swap out the `MemcachedJumpHashSelector` with a mock so that I could test the `sortKeysByServer` method. I will add a comment to this method based on the existing comment on `MemcachedJumpHashSelector.SetServers`